.github/workflows/ci.yml

```
@@ -0,0 +1,376 @@
+        # Basic functionality test
+        echo "Testing basic startup and shutdown..."
+        ls -la ./vsg_ode_robot
+        ldd ./vsg_ode_robot || true
+        timeout 15s ./vsg_ode_robot || echo "Basic integration test completed"
+        
+        # Memory leak check (short run)
+        echo "Running memory leak detection..."
+        timeout 20s valgrind --leak-check=full --show-leak-kinds=all \
+          --error-exitcode=1 --track-origins=yes \
+          ./vsg_ode_robot 2>&1 | tee valgrind.log || echo "Memory test completed"
+        
```

YAMLlint (1.37.1)

[error] trailing spaces

(trailing-spaces)